test/connectFactoryObservable.test.tsx

```diff
@@ -41,4 +42,72 @@ describe("connectObservable", () => {
     renderHook(() => useLatestNumber(6))
     expect(nInitCount).toBe(2)
   })
+
+  test("getObservable can take an object as the key", async () => {
+    const obj1 = {
+      foo: "foo",
+    }
+    const obj2 = {
+      foo: "foo2",
+    }
+    const obj3 = {
+      foo: "foo3",
+    }
+
+    let count = 0
+
+    const [useFoo] = connectFactoryObservable((obj: { foo: string }) =>
+      defer(() => {
+        count++
+        return of(obj.foo).pipe(map(text => [text, count].join(".")))
+      }),
```

Comment:
That's a nice way of asserting recalculations. I like it 👍

Reply:
:see_no_evil: the tests are horrible because they are not descriptive and they test way too much... However, they serve the purpose of making sure that I don't break anything major after every refactor, and as ugly as they are they have prevented me from some pretty shitty things already. Anyways, I'm currently working on a better set of tests.